api/src/main/java/au/org/aodn/nrmn/restapi/controller/StagedJobController.java

```diff
@@ -378,4 +381,25 @@ public void downloadFile(final HttpServletResponse response, @PathVariable Strin
                     jobId, clientException.getMessage())));
         }
     }
+
+    private void filterNonBottomDuplicateRows(Collection<Long> rowIdsToRemove, Long lastRowId) {
+
+        var toRemoveIds = new ArrayList<Long>();
+
+        if (!rowIdsToRemove.contains(lastRowId)) {
```

Comment:
Is this a reliable check, can the lastRowId not empty and then it will not exist in the rowIdsToRemove()?

Reply:
Sorry I didn't get your point. The meaning of this part is: Because we only want to remove duplicate rows on the bottom. So, if the `rowIdsToRemove` doesn't contain the last row, it means no rows should be removed.